service/history/shard/controller_impl.go

```diff
@@ -399,13 +400,15 @@ func (c *ControllerImpl) acquireShards() {
 			return
 		}
 
+		systemBackgroundCtx := headers.SetCallerInfo(context.Background(), headers.SystemBackgroundCallerInfo)
```

Comment:
Does this caller has the top priority?

Reply:
no, but it doesn't matter. Shard won't move or renew if `AssertOwnership` fails.